src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/command/flow/ingress/of/IngressInstallFlowModFactory.java

```diff
@@ -107,7 +107,10 @@ public IngressInstallFlowModFactory(
         return instructions;
     }
 
-    protected abstract List<OFAction> makeServer42IngressFlowTransformActions(List<Integer> vlanStack);
+    protected List<OFAction> makeServer42IngressFlowTransformActions(List<Integer> vlanStack) {
+        // default implimentation to do not force FlowLoop and OneSwitchFlow factories to override this method
+        return new ArrayList<>();
```

Comment:
Safer to return empty list in class that do not need implementantion, than do raturn empty list in class that must implement it.